source/MaterialXRender/Mesh.cpp

```diff
@@ -205,4 +205,34 @@ void Mesh::splitByUdims()
     }
 }
 
+void MeshStream::transform(const Matrix44 &matrix)
+{
+    unsigned int stride = getStride();
+    size_t numElements = _data.size() / getStride();
+    for (size_t i=0; i<numElements; i++)
+    {
+        Vector4 vec(0.0, 0.0, 0.0, 1.0);
+        for(size_t j=0; j<stride; j++)
+        {
+            vec[j] = _data[i*stride + j];
+        }
+        if(getType() == MeshStream::POSITION_ATTRIBUTE ||
+           getType() == MeshStream::TEXCOORD_ATTRIBUTE ||
+           getType() == MeshStream::GEOMETRY_PROPERTY_ATTRIBUTE)
```

Comment:
As a note: We don't have the concept of a vector vs position property so we can assume positional for now. I'm not sure if geomattrvalue has this concept or not in any case.